### ISSUE TITLE
Resolving the issue of black video

### ISSUE THREAD

**Issue by dec05eba** [HoroTW/GPU-Screen-Recorder#3]
Hi, I found this repo by accident and since nobody reported these issues to me I didn't know that they existed (I dont use replay 24/7 and I dont use a screen saver, etc).
So I'm wondering, are there any other scenarios in which this happens other than sleep/screen saver/hibernation (graphics context lost) or when monitor setup is reconfigurated?
I will fix it properly if I know in which cases it can happen.

**Comment by dec05eba** [comment]
Ok so, hibernate issue cant be fixed. It's a NVIDIA driver bug with CUDA. If CUDA is running when suspend/hibernate happens then CUDA breaks and the only way to fix it is to do rmmod nvidia_uvm, modprobe nvidia_uvm. Seems like the user has to set the NVreg_PreserveVideoMemoryAllocations=1 option :(
But the other issues should be fixable.

**Comment by HoroTW** [comment]
Hey dec05eba :)
I would have eventually wrote you an email, when I was done with finding out what I could recommend...
Since then I found an really annoying bug in the NVIDIA Driver....
First things first: I had 24/7 recording working fine for over a week (which was amazing and I already started writing you)... but then I found that using the NVENV/NVDEC (which uses CUDA) the GPU is hanging in a Power state that draws 70-100W more than IDLE ....

see https://github.com/NVIDIA/open-gpu-kernel-modules/issues/333

This is not too bad when Gaming... (I saw you also found the CUDA memory frequency limitation) ... Yeah this is somewhat related to that I guess...

I confirmed that on Windows the shadow replay is working without kicking the GPU in high gear... But I did not yet find a way (besides hard limiting the frequency) to workaround this limitation.....

Maybe you know / will find something? I will happily help where I can :)
I wanted to check if in a GPU limiting (due to thermal throttling) Game I would find a loss in performance. Alternatively I will check the consumption in a game where the GPU is in the same or higher power state.

For now I feel for a 24/7 recording CPU encoding is superior... which is such a shame .... I really like this project :/

Greetings ~Horo

**Comment by dec05eba** [comment]
Hmm I see, yeah I have been trying to find a way to disable that "force p2 state". There is a way to do it on windows with nvapi but nvidia has not made nvapi available for linux. Nvidia has a similar profile option on linux to set application specific driver settings but that "force p2 state" option doesn't seem to be available there and it seems like nobody has found a way to disable that driver "feature" (bug). I guess I'll have to reverse engineer the driver to see how it turns on that "force p2 state" on linux, since it seems like nvidia only turns it off for proprietary software, on windows :(

**Comment by HoroTW** [comment]
Yeah, I came to the same conclusion but since I don't have much experience in that field I'm unsure how to start... 
I would like to help you, but I'm not sure if I can :/
